src/wagtailtrans/views/language.py

```diff
@@ -0,0 +1,23 @@
+from __future__ import absolute_import, unicode_literals
+
+from wagtail.contrib.modeladmin.views import DeleteView
+
+
+class LanguageDeleteView(DeleteView):
+    """
+    Custom Delete View Class for `Language` ModelAdmin.
+    """
+    def post(self, request, *args, **kwargs):
+        """Overriding Default post method because we need to delete related
+        translatable pages (if there is any) before we can delete a Language
+        which are references through protected foreign key.
+
+        Also we'll not allow to delete the default language.
+        """
+        if self.instance.is_default:
+            raise Exception("Can't delete a default language")
+
+        if self.instance.pages.count():
+            self.instance.pages.all().delete()
```

Comment:
You don't have to change this, but why not make the foreign key in TranslatablePage a `on_delete=models.CASCADE`? I'm curious about your thoughts on that.

Reply:
As incase of `sync=True`,  `on_delete=models.CASCADE` will make our tree unstable, I think it's better to handle it here.